### DISCUSSION TITLE
Customize test_case id for Junit reporting

### DISCUSSION THREAD

**Discussion by ss-napier** [seleniumbase/SeleniumBase#2743] [Q&A] [answered]
Hey there, 

Can you help to suggest idea how can we customize test_case id without forking the project especially when using --junit-xml parameter?

thanks!

**Comment by mdmintz** [comment, marked as answer]
`--junit-xml` comes from `pytest`, not `seleniumbase`.

https://github.com/pytest-dev/pytest/blob/97610067acf220b13cc4c6fca34e5aac4834b04d/src/_pytest/junitxml.py#L387

**Reply by ss-napier** [reply]
okey understand, but then how can we define our own test_id. I notice there is self.test_id through BaseCase.

**Reply by mdmintz** [reply]
The `test_id` is special. It's how `pytest` locates your test. It's based on the file path, class name, and test name. If you want a different ID, you'll need to change the name of the file / class / test method.
